### PR TITLE
Feature/robot  pmtalloc automation

### PR DESCRIPTION
**Pull request by skristem**
# Critical Changes

# Changes

# Issues Closed

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by davidmreed** [comment]
@skristem Do you want to merge https://github.com/SalesforceFoundation/Cumulus/pull/4378 into your branch to provide metadata?

File: robot/Cumulus/tests/browser/zpayment_allocations/enable_pmt_alloc_and_test.robot [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    &{def_gau} =  API Create GAU    Name=default gau
+    Set Global Variable     &{def_gau}    &{def_gau}
+    Go To Setup Page    CustomSettings
+    Select Frame And Click Element    Custom Settings    custom_settings.link    Allocations Settings    Manage
+    Unselect Frame
+    Wait Until Loading Is Complete
+    Choose Frame    Custom Setting Allocations Settings
+    Wait Until Element Is Visible  text:Default Organization Level Value
+    Checkbox Status    Payment Allocations Enabled    Not Checked
+    Checkbox Status    Default Allocations Enabled    Not Checked
```
Comment (Safiyab) [inline robot/Cumulus/tests/browser/zpayment_allocations/enable_pmt_alloc_and_test.robot]: The only differences I see for the keywords 'Enable payment allocations' and Disable payment allocations' are
- Checkbox statuses
- Populating the Default__c field

Can we not create just one keyword for these both and pass the status and default__c value as the parameters ? And you can  move the creation of &{def_gau} to 'Setup test data' keyword itself.

File: unpackaged/config/qa/package.xml [hunk 2]
```diff
@@ -47,6 +47,7 @@
     </types>
     <types>
         <members>Contact-Contact Lightning Layout</members>
+        <members>npe01__OppPayment__c-npe01__Payment Layout</members>
```
Comment (Safiyab) [inline unpackaged/config/qa/package.xml]: Do we commit the changes to "package.xml" file as part of the retrieve_changes task ? I'll let @davidmreed confirm.

File: robot/Cumulus/resources/NPSP.py [hunk 3]
```diff
@@ -124,12 +124,14 @@ def select_tab(self, title):
...
-                button.click()
-                time.sleep(5)
-                tab_found = True
-                break
+                buttons = self.selenium.get_webelements(locator)
+                for button in buttons:
+                    if button.is_displayed():
+                        self.salesforce._focus(button)
+                        button.click()
+                        time.sleep(5)
```
[the remark this answers is not in the document]
Comment (skristem) [reply robot/Cumulus/resources/NPSP.py]: In some cases the page takes lot of time to load so i have this 5 seconds sleep and this keyword is used in almost every test case

File: robot/Cumulus/tests/browser/zpayment_allocations/enable_pmt_alloc_and_test.robot (hunk 1 above)
Comment (skristem) [reply robot/Cumulus/tests/browser/zpayment_allocations/enable_pmt_alloc_and_test.robot]: I have to create gau as part of enabling as I need to provide gau id while enabling. And my other test can only be created after the GAU is enabled. I am adding GAU as part of enabling apart from checkbox and removing GAU as part of disabling. So I cannot combine these keywords

**Comment by davidmreed** [comment]
@skristem The Robot test build is failing - could you take a look?

File: unpackaged/config/qa/package.xml (hunk 2 above)
Comment (davidmreed) [reply unpackaged/config/qa/package.xml]: Yeah, we need this here.

**Pull request merged**